openhands/events/stream.py

```diff
@@ -138,6 +138,10 @@ def add_event(self, event: Event, source: EventSource):
             asyncio.run(self.async_add_event(event, source))
 
     async def async_add_event(self, event: Event, source: EventSource):
+        if hasattr(event, '_id') and event.id is not None:
+            raise ValueError(
+                'Event already has an ID. It was probably added back to the EventStream from inside a handler, trigging a loop.'
+            )
```

Comment:
This should work for that case! Although it will fail hard with ValueError, and the FatalObs is in the eventstream, but I think evals don't access the stream directly.

If so, maybe we can add a `cause` to this or something, so that it can be discovered on the other side and reported?

Reply:
Actually, I don't know, can you please point out where is it sending the same? I see `send_error` is creating another ErrorObs. 🤔

Reply:
Ohh I think you're right actually :/

It's late. Going to switch this to draft